enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/adm/VulnerabilityWorker.java

```diff
@@ -569,14 +575,19 @@ private int convert(Dependency dependency, Map<String, Integer> gavIndex, List<A
         List<String> childrenNodeIds = new ArrayList<>(dependency.getChildren().size());
         for (Dependency childDependency : dependency.getChildren()) {
             int cid = convert(childDependency, gavIndex, result);
-            childrenNodeIds.add(Integer.toString(cid));
+            if (cid != -1) {
+                childrenNodeIds.add(Integer.toString(cid));
+            }
         }
         builder.applicationDependencyNodeIds(childrenNodeIds);
         result.add(builder.build());
         return n;
     }
 
     private static String createGAV(ArtifactSpec artifact) {
+        if (artifact == null) {
+            return null;
+        }
         StringBuffer sb = new StringBuffer();
```

Comment:
how about movig to StringBuilder and initializing it's capacity to avoid inner buffer reallocations?

Reply:
IMHO not worth the extra computation; but StringBuilder is good. Will change - thanks/

Reply:
You dont have to calulate it. Just use a reasonable constant like new StringBuilder(150) or new StringBuilder(200) which will be enough most of the time.

Reply:
To be honest, alhtough I think that great things are made from very small ones,  I think it's a pity the review focuses just on such nitpicking as this conversation. I am sure there are larger either structural or conceptual mistakes in my commits that would deserve a thorough review.

In itself, this wouldn't make me to make another commit, but as there are other relevant  code sins pointed out, will fix.

Reply:
addressed in 174665381875

Reply:
That's bad. 
Speaking for myself, I can't promise I am 'not one of them', but at least I can TRY not be one of them - no guarantee on the outcome. Try me ;)

Other thought: consider (just) the size of the codebase and average age of the code (= techniques which were valid 15 yrs ago, but are obsolete now). Then weight in the benefit of a refactoring vs. the possibility of bugs slipping in. This comparison may easily result in a rejection for reasons that may seem silly and/or evasive to new contributors. Just a thought. Anyway, any such feelings should be voiced on the mailing list.

Reply:
"weight in the benefit of a refactoring vs. the possibility of bugs slipping in."
That's why unit test got invented ... oh wait ... unit test PRs get ignored as well.